### PR TITLE
Windows compatability

### PR DESCRIPTION
**Pull request by jwfearn**
For this project to work better on Windows, a few minor changes are needed:
- Don't use `bin/rake` stubs created by Bundler's `--binstubs` option.  The stubs are a handy convenience to save typing but sadly they only work on Mac/Linux.
- Use a newer version of the `yard` gem

Also added some tips on Ruby installation to the README.

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -38,9 +44,9 @@ bootstrapped:
 
     git clone git://github.com/rspec/rspec-dev.git
     cd rspec-dev
-    bundle install --binstubs
-    bin/rake setup
-    bin/rake # runs tests in every repository
+    bundle install --path=vendor/bundle
```
Comment (myronmarston) [inline README.markdown]: Why the change here and the specified `--path`?  Users can use any `--path` they want (or none).  That's a personal setting and not something that seems to warrant a recommendation here.

File: Rakefile [hunk 2]
```diff
@@ -347,13 +347,13 @@ end
...
 task :default do
-  run_command "bin/rake"
+  run_command "bundle exec rake"
 end
 
 desc "publish cukes to relishapp.com"
 task :relish, :version do |_, args|
   raise "rake relish[VERSION]" unless args[:version]
-  run_command "bin/rake relish['#{args[:version]}']", :except => ['rspec']
+  run_command "bundle exec rake relish['#{args[:version]}']", :except => ['rspec']
```
Comment (myronmarston) [inline Rakefile]: Did windows have problems with `bin/rake` for some reason?

FWIW, most of these tasks are things that only I use (e.g. to release new versions, update the relish docs, etc) and I prefer the binstubs approach.  I'd prefer to stick with `bin/rake`.  If windows needs it to be `bundle exec`, maybe we can put something like this at the start of the rake file?

``` ruby
RAKE_COMMAND = if RbConfig::CONFIG['host_os'] =~ /cygwin|mswin|mingw|bccwin|wince|emx/
  "bundle exec rake"
else
  "bin/rake"
end
```

...and then we can use `RAKE_COMMAND` in all these places -- that way, it'll use `bundle exec rake` on windows, and `bin/rake` elsewhere.

File: README.markdown (hunk 1 above)
Comment (jwfearn) [inline README.markdown]: Agreed.  But since the rest of the text recommends isolating gems, that's a way to do it that doesn't depend on "Gemsets" which AFAIK is a feature of RVM and perhaps some other Rbuy managers.  But not `uru` and maybe not `rbenv` either.

File: Rakefile (hunk 2 above)
Comment (jwfearn) [inline Rakefile]: Yes, the binstubs create by Bundler rely on shebangs to run so they are Linux/Mac only.  There is a gem that will produce Windows-compatible stubs but I was reluctant to add a gem dependency when there was a simple, cross platform, solution.

I guess one could add `RAKE_COMMAND` as you suggest but it just seems like additional complexity.

One could still use `--binstubs` and `bin/rake`, `bin/yard`, etc. from the command line.  Only in scripts would it need to change.

File: README.markdown (hunk 1 above)
Comment (jwfearn) [inline README.markdown]: That said maybe I should either add an explanation about `--path=vendor/bundle` or leave it out.  And running with `--binstubs` on Windows doesn't fail, it's just that the stubs themselves won't work.

Comment (JonRowe) [inline README.markdown]: Leave it out please, it actually forces a convention onto people they may not follow, I'd actually also rather add the gem for windows users to enable compatibility with bin stubs.

Comment (jwfearn) [inline README.markdown]: I just tried the `rubygems-bundler` gem (suggested in [this two-year-old stackoverflow answer](http://stackoverflow.com/questions/11922386/railss-bundle-binstubs-on-windows)).  It doesn't work.

**Comment by benoittgt** [comment]
I think this PR is too old and should be closed.
